src/ixdat/readers/cinfdata_db.py

```diff
@@ -0,0 +1,411 @@
+        a setup at SurfCat at given timestamp
+
+        All attributes of this reader can be accessed from the
+        measurement as `measurement.reader.attribute_name`.
+
+        Args:
+            path_to_file (str): Named argument from Measurement Class.
+                Can be used as the setup name in the cinfdatabase
+            **kwargs (dict): Key-word arguments are passed to cinf Measurement.__init__
+                setup_name (str): The setup name in the database default to path_to_file
+                timestamp (str): Timestamp the measurement started given as
+                    (YYYY-MM-DD HH:MM:SS)
+        """
+
+        self.measurement_class = kwargs.pop("measurement_class", cls)
+        self.setup_name = kwargs.pop("setup_name", path_to_file)
+        self.timestamp = kwargs.pop("timestamp", None)
+        self.comment = kwargs.pop("comment", None)
+        self.mass_scans = kwargs.pop("include_mass_scans", False)
+        self.verbose = kwargs.pop("verbose", False)
+        self.grouping_column = kwargs.pop("group", None)
+
+        # figure out whether to collect data group by 'comment' or 'timestamp'
+        if not self.grouping_column:
+            if self.timestamp and not self.comment:
+                self.grouping_column = "time"
+                self.token = self.timestamp
+            elif self.comment and not self.timestamp:
+                self.grouping_column = "comment"
+                self.token = self.comment
+            else:
+                warnings.warn(
+                    "Both a comment and a timestamp is given "
+                    "but no explicit grouping column is set. \n"
+                    f"Defaults to 'time' using '{self.timestamp}'",
+                    stacklevel=2,
+                )
+                self.grouping_column = "time"
+                self.token = self.timestamp
+
+
+        if issubclass(self.measurement_class, Spectrum):
+            return self.read_spectrums()
+
+        elif issubclass(self.measurement_class, Measurement):
+            obj_as_dict = self.read_ms()
+            if issubclass(cls, self.measurement_class):
+                self.measurement_class = cls
+            obj_as_dict.update(kwargs)
+            self.measurement = self.measurement_class.from_dict(obj_as_dict)
+
+            if self.mass_scans:
+                if self.verbose:
+                    print("adding mass scans to the measurement")
+                self.measurement = self.add_mass_scans()
+
+            # self.data_has_been_fetch = True
+
+            return self.measurement
+
+    def read_ms(self):
+        """Download MS data from cinfdata_database"""
+
+        self.cinf_db = plugins.cinfdata.connect(
+            setup_name=self.setup_name, grouping_column=self.grouping_column
+        )
+
+        self.group_data = self.cinf_db.get_data_group(
+            self.token, scaling_factors=(1e-3, None)
```

Comment:
This value is used to scale the time values into seconds instead of milliseconds original from cinfdata. 

@ScottSoren  Maybe this should be taken out and scaled in ixdat with the pint unit library later when plotting.  

I will make it a module constant for now.